python/packages/nisar/workflows/rubbersheet.py

```diff
@@ -22,9 +24,197 @@
+    '''
+    # Pull parameters from cfg dictionary
+    ref_hdf5 = cfg['input_file_group']['reference_rslc_file']
+    scratch_path = pathlib.Path(cfg['product_path_group']['scratch_path'])
+    rubbersheet_params = cfg['processing']['rubbersheet']
+    geo2rdr_offsets_path = pathlib.Path(rubbersheet_params['geo2rdr_offsets_path'])
+    off_product_enabled = cfg['processing']['offsets_product']['enabled']
+    dem_file = cfg['dynamic_ancillary_file_group']['dem_file']
+
+    # If not set, set output HDF5 file
+    if output_hdf5 is None:
+        output_hdf5 = cfg['product_path_group']['sas_output_file']
+
+    info_channel = journal.info('rubbersheet.run_rubbersheet_with_polyfit')
+    t_all = time.time()
+
+    ref_slc = SLC(hdf5file=ref_hdf5)
+
+    with HDF5OptimizedReader(name=output_hdf5, mode='r+',
+                             libver='latest', swmr=True) as dst_h5:
+        for freq, _, pol_list in get_cfg_freq_pols(cfg):
+            # Get the slant range and zero doppler time spacing
+            ref_radar_grid = ref_slc.getRadarGrid(freq)
+            ref_slant_range_spacing = ref_radar_grid.range_pixel_spacing
+            ref_zero_doppler_time_spacing = ref_radar_grid.az_time_interval
+
+            prf = ref_radar_grid.prf
+            rsr = isce3.core.speed_of_light/ref_slant_range_spacing * 0.5
+            rbw = ref_slc.getSwathMetadata(freq).processed_range_bandwidth
+            abw = ref_slc.getSwathMetadata(freq).processed_azimuth_bandwidth
+
+            ref_az_times = ref_slc.getZeroDopplerTime()
+            ref_slant_ranges = ref_slc.getSlantRange(freq)
+            minL, maxL, minP, maxP = 0, len(ref_az_times), 0, len(ref_slant_ranges)
+
+            freq_group_path = f'{RIFGGroupsPaths().SwathsPath}/frequency{freq}'
+            pixel_offsets_path = f'{freq_group_path}/pixelOffsets'
+            geo_offset_dir = geo2rdr_offsets_path / 'geo2rdr' / f'freq{freq}'
+            rubbersheet_dir = scratch_path / 'rubbersheet_offsets' / f'freq{freq}'
+            off_slant_range = dst_h5[f'{pixel_offsets_path}/slantRange'][()]
+            off_zero_doppler_time = dst_h5[f'{pixel_offsets_path}/zeroDopplerTime'][()]
+
+            off_az_indices = ((off_zero_doppler_time - ref_az_times[0]) *
+                              ref_radar_grid.prf).astype(int)
+            off_rg_indices = ((off_slant_range - ref_slant_ranges[0])/
+                              ref_slant_range_spacing).astype(int)
+
+            # Produce ground track velocity for the frequency under processing
+            ground_track_velocity_file = get_ground_track_velocity_product(ref_slc,
+                                                                           off_slant_range,
+                                                                           off_zero_doppler_time,
+                                                                           dem_file,
+                                                                           rubbersheet_dir)
```

Comment:
```suggestion
            ground_track_velocity_file = get_ground_track_velocity_product(
                ref_slc,
                off_slant_range,
                off_zero_doppler_time,
                dem_file,
                rubbersheet_dir,
            )
```
This is a war that I will probably never win, but these parameter lists that are way up to the right are more difficult to read. Also, it's best to use keyword arguments where possible so that if the order of function parameters changes in the function being called, the code is less likely to break or, if it does, it will at least give a more appropriate error.